templates/ServiceClientFile.stencil

```diff
@@ -31,6 +31,9 @@ public final class {{ model.name }}: {{ model.protocols }} {
     public let options: {{ model.name }}Options
 
     // MARK: Initializers
+    {% if model.host != "" %}
+    {% include "ClientConvenienceInitSnippet.stencil" %}
+    {% endif %}
```

Comment:
Could we not simply provide a default value for `baseUrl` in the regular init if the host parameter is supplied?

```swift
public init(
  baseUrl: URL = URL("blah"),
  authPolicy: Authenticating,
  withOptions options: Options
) throws {
```

Reply:
This does not work because URL("blah") will return an optional URL object. Therefore, we need to unwrap it because we can use it.

Reply:
In that case, we can do this:

```swift
public init(
  baseUrl: URL? = nil,
  authPolicy: Authenticating,
  withOptions options: Options
) throws {
let hostValue = baseUrl ?? URL(String: hostValue)
guard hostValue != nil else {
  logger.error("Must supply URL or hostValue")
}
```

Something like that.

Reply:
If $host is provided, is it provided per operation, or for the whole client? If for the whole client, should it not simply be a constant on the client?

Reply:
It is for the whole client. I am applying the whole constant in the convenience init(). I just create an URL object from the constant add a guard before calling the init(). This is an example of the generated code.

```
   public convenience init(
        authPolicy: Authenticating,
        withOptions options: AutoRestSwaggerBatFileClientOptions
    ) throws {
        guard let baseUrl = URL(string: "http://localhost:3000") else {
            fatalError("Unable to form base URL")
        }
        try self.init(
            baseUrl: baseUrl,
            authPolicy: authPolicy,
            withOptions: options
        )
    }
```

Reply:
```swift
public init(
        baseUrl: URL? = nil,
        authPolicy: Authenticating,
        withOptions options: AutoRestSwaggerBatFileClientOptions
    ) throws {
        let defaultHost = URL(string: "http://localhost:3000")
        let host = baseUrl ?? defaultHost
        guard host != nil else {
            fatalError("Unable to determine base URL. ")
        }
        try self.init(
            baseUrl: baseUrl,
            authPolicy: authPolicy,
            withOptions: options
        )
    }
```

If the swagger doesn't designate a $host property, then the initializer can be generated the same as it currently does. What I would like to avoid is another stencil file that essentially duplicates code from elsewhere.

Reply:
I was trying to avoid adding a if logic to swagger , that is why i create a new swagger for this scenario. 
Regardless, thanks for the feedback, I have upated new iteration per feedback.